tools/ARIAtools/product.py

```diff
@@ -321,7 +329,9 @@ def __init__(self, filearg, bbox=None, workdir='./', num_threads=1,
                     'products: Linux kernel >=4.3 and libnetcdf >=4.5')
 
         # check if local file reader is being captured as netcdf
-        if any("https://" not in i for i in self.files):
+        check_for_urls = any("https://" not in i for i in self.files)
```

Comment:
What is being determined by these lines of code? If this is a nisar file? If the file ends with .h5? If the file starts with https://?

IMO this can be made much more clear / lower cognitive load